src/api/api_core.php

```diff
@@ -54,13 +54,9 @@ function login($username, $password) {
             $query->execute(array(':id'       => $result['id'],
                                   ':auth_key' => $auth_key));
 
-            // set authorisation cookie
-            setcookie('auth_userid', $result['id'], 0, '/', '', FALSE);
-            setcookie('auth_key', $auth_key, 0, '/', '', FALSE);
-            session_regenerate_id(TRUE);
-            $_SESSION['user_id'] = $result['id'];
-            $_SESSION['user_name'] = $result['name_ingame'];
-            $_SESSION['user_lastactive'] = time();
+            update_session($result['id'], $result['name_ingame']);
+            set_authorisation_cookies($result['id'], $auth_key, $doStayLoggedIn);
```

Comment:
Hmm, so in the previous code, we set the user's cookies first, and set the current session variables second.  Now the order is reversed.  Any particular reason?  I don't offhand see a problem necessarily (all of this state is stuff that belongs to the user that we're not validating or trying to store in a database), but i need to think about it a little more.